File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int start = 0;
        int end = arr.length - 1;

        while(start <= end) {
            int mid = (start + end) / 2;
            
                if(mid - 1 >= 0 && mid + 1 < arr.length && arr[mid - 1] < arr      [mid] && arr[mid] > arr[mid + 1]) {
                    return mid;
                }
            
                else if(mid - 1 < 0 || arr[mid] > arr[mid - 1]) {
                    start = mid + 1;
                }

                else if(mid + 1 >= arr.length || arr[mid + 1] < arr[mid]) {
                    end = mid - 1;
                }
            }
        
        return 0;
    }
}

File: 0882-peak-index-in-a-mountain-array/0882-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int start=0,end=arr.size()-1;
        int mid=0;
        while(start<=end)
        {
            mid=start+(end-start)/2;
            if(mid!=0 && mid!=arr.size()-1)
            {

            if(arr[mid]>arr[mid+1] && arr[mid]>arr[mid-1])
            {
                return mid;
            }
            else if(arr[mid]<arr[mid-1])
            end=mid-1;
            else if(arr[mid]>arr[mid-1])
            start=mid+1;
            }
            // if (mid==start)
            // return mid;
            // else if(mid==end)
            // return mid;
            if(mid==0)
            return mid+1;
            else if(mid==end)
            return mid-1;

        }
        return 0;
    }
};